File: ainpc-core-plugin/src/main/java/ro/ainpc/engine/runtime/ScenarioValidationReport.java
package ro.ainpc.engine.runtime;

import java.util.ArrayList;
import java.util.List;

public class ScenarioValidationReport {

    private final List<String> errors = new ArrayList<>();
    private final List<String> warnings = new ArrayList<>();
    private final List<String> infos = new ArrayList<>();

    public void error(String message) {
        add(errors, message);
    }

    public void warn(String message) {
        add(warnings, message);
    }

    public void info(String message) {
        add(infos, message);
    }

    public boolean valid() {
        return errors.isEmpty();
    }

    public boolean isValid() {
        return valid();
    }

    public boolean hasWarnings() {
        return !warnings.isEmpty();
    }

    public void merge(ScenarioValidationReport other) {
        if (other == null) {
            return;
        }
        errors.addAll(other.errors());
        warnings.addAll(other.warnings());
        infos.addAll(other.infos());
    }

    public List<String> errors() {
        return List.copyOf(errors);
    }

    public List<String> warnings() {
        return List.copyOf(warnings);
    }

    public List<String> infos() {
        return List.copyOf(infos);
    }

    private void add(List<String> target, String message) {
        if (message != null && !message.isBlank()) {
            target.add(message.trim());
        }
    }
}

File: ainpc-core-plugin/bin/main/ro/ainpc/engine/runtime/ScenarioValidationReport.kt
package ro.ainpc.engine.runtime

class ScenarioValidationReport {
    private val errors = ArrayList<String>()
    private val warnings = ArrayList<String>()
    private val infos = ArrayList<String>()

    fun error(message: String?) {
        add(errors, message)
    }

    fun warn(message: String?) {
        add(warnings, message)
    }

    fun info(message: String?) {
        add(infos, message)
    }

    fun valid(): Boolean = errors.isEmpty()

    fun isValid(): Boolean = valid()

    fun hasWarnings(): Boolean = warnings.isNotEmpty()

    fun merge(other: ScenarioValidationReport?) {
        if (other == null) {
            return
        }
        errors.addAll(other.errors())
        warnings.addAll(other.warnings())
        infos.addAll(other.infos())
    }

    fun errors(): List<String> = errors.toList()

    fun warnings(): List<String> = warnings.toList()

    fun infos(): List<String> = infos.toList()

    private fun add(target: MutableList<String>, message: String?) {
        if (!message.isNullOrBlank()) {
            target.add(message.trim())
        }
    }
}
